### PR TITLE
performance improvement in Database\BaseResult to use truthy check instead of count($var) when possible

### PR DESCRIPTION
**Pull request by samsonasik**
Applied to `resultObject` and `resultArray`:

Before:

```php
if (($c = count($this->resultObject)) > 0)
{
    for ($i = 0; $i < $c; $i ++ )
    {
    }
}
```

After:

```php
if ($this->resultObject)
{
    foreach ($this->resultObject as $row)
    {
    }
}
```

**Checklist:**
- [x] Securely signed commits
- [x] Conforms to style guide

### CODE REVIEW

File: system/Database/BaseResult.php [hunk 1]
```diff
@@ -213,16 +213,16 @@ public function getResultArray(): array
...
 		}
 
-		if (($c = count($this->resultObject)) > 0)
+		if ($this->resultObject)
 		{
-			for ($i = 0; $i < $c; $i ++ )
+			foreach ($this->resultObject as $row)
 			{
-				$this->resultArray[$i] = (array) $this->resultObject[$i];
+				$this->resultArray[$i] = (array) $row;
```
Comment (puschie286) [inline system/Database/BaseResult.php]: you forgot $i here

Comment (puschie286) [reply system/Database/BaseResult.php]: i mean there should be no $i in this function

Comment (puschie286) [reply system/Database/BaseResult.php]: dont know why the Travis build success but you forgot to remove the $i on the left hand array in both foreach loops ( l:225, l:268 )

Comment (samsonasik) [reply system/Database/BaseResult.php]: Ooops, fixed now. Thank you @puschie286

**Pull request merged**